appeals/web/src/server/appeals/appeal-details/appellant-case/__tests__/appellant-case.test.js

```diff
@@ -1162,6 +1163,212 @@ describe('appellant-case', () => {
+			await request.post(`${baseUrl}/1${appellantCasePagePath}`).send({
+				reviewOutcome: 'valid'
+			});
+
+			const response = await request.get(
+				`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`
+			);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+	});
+
+	describe('POST /appellant-case/valid/date', () => {
+		beforeEach(async () => {
+			await request.post(`${baseUrl}/1${appellantCasePagePath}`).send({
+				reviewOutcome: 'valid'
+			});
+		});
+
+		afterEach(() => {
+			nock.cleanAll();
+		});
+
+		it('should re-render the valid date page with the expected error message if no date was provided', async () => {
+			const response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '',
+					'due-date-month': '',
+					'due-date-year': ''
+				});
+
+			expect(response.statusCode).toBe(200);
+			const element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the valid date page with the expected error message if provided date is not in the future', async () => {
+			const response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '1',
+					'due-date-month': '1',
+					'due-date-year': '2000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			const element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the valid date page with the expected error message if an invalid day was provided', async () => {
+			let response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '0',
+					'due-date-month': '1',
+					'due-date-year': '3000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			let element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+
+			response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '32',
+					'due-date-month': '1',
+					'due-date-year': '3000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+
+			response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': 'first',
+					'due-date-month': '1',
+					'due-date-year': '3000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the update date page with the expected error message if an invalid month was provided', async () => {
```

Comment:
Would it be worth breaking the test into parts specific to what it is testing? This would avoid overlap.